### ISSUE TITLE
Expected Artifact Http -default artifact URL field in UI saves to "name" instead of "reference" in JSON

### ISSUE THREAD

**Issue by away168** [spinnaker/spinnaker#4395]
### Issue Summary:
Expected Artifact kind: http - 
URL field for Default Artifact records the contents into "name" instead of "reference" 

### Cloud Provider(s):
Any

### Environment:
Spinnaker v 1.13.4

### Feature Area (if this issue is UI/UX related, please tag `@spinnaker/ui-ux-team`):
@spinnaker/ui-ux-team

### Description:
When configuring an Expected Artifact of kind: HTTP, and specifying the "Default Artifact" as also kind: HTTP, the URL field gets recorded as "name" within JSON instead of "reference"

### Steps to Reproduce:
Configure stage. Add an Expected Artifact
Kind: HTTP
Check - Use Default Artifact
Kind: HTTP for Default Artifact
URL: http://example.com/myfile.yml
Save
Show Pipeline as Json

### Additional Details:

Sample JSON:

```json
"expectedArtifacts": [
   {
     "defaultArtifact": {
       "id": "570894a9-a708-4bee-b59b-fb2b49b2e8f7",
       "name": "https://example.com/myfile.yml",
       "type": "http/file"
     },
     "displayName": "myhttpartifact",
     "id": "a5e158b0-4c93-4998-ae8b-4b6856042b38",
     "matchArtifact": {
       "id": "7203659f-6f0f-45cc-8e6c-d5ff6f67f24d",
       "reference": "https://example.com/myfile.yml",
       "type": "http/file"
     },
     "useDefaultArtifact": true,
     "usePriorArtifact": false
   }
 ],
```

**Comment by away168** [comment]
Workaround is simply to edit the json and change the field (under "defaultArtifact") from "name" to "reference"

**Comment by ezimanyi** [comment]
This was fixed in spinnaker/deck#6836; I've just cherry-picked the fix into 1.13 so it will be in the next 1.13.x patch release (and also in 1.14 of course).
